### ISSUE TITLE
Recommended approach to install language packs

### ISSUE THREAD

**Issue by smejkal** [IBM/domino-docker#55]
What is the recommended approach to install language packs with Domino on Docker?

Currently I would do the following:

- untar language pack in local directory
- create LPSilent.ini in same directory
- stop domino container
- start temporary container and map volume /local/notesdata/ and install folder in /tmp/install
- cd /tmp/install
- ./LNXDomLP110 -i silent -DSILENT_INI_PATH="/tmp/install/LPSilent.ini"
- alternatively: start ./LNXDomLP110 without LPSilent.ini

Is there a better approach?

**Comment by Daniel-Nashed** [comment]
Do you really need a LP? What is the scenario? A first server? Additional servers?
Which databases need to be localized in your case?

Over all the years you are the 3rd person asking. So I did not spend any time on it yet.

The right way would be to add it to the software file, but then we would need to support all the languages...

I would take the files from another machine and don't manage it via the container.

your approach would be the right approach. But I can't spend time on supporting you. 

are you planning to build your add-on image?

I would keep system databases in English always and address mail-files and other templates separately for the whole Domain.

Which language would you need?

**Comment by smejkal** [comment]
No, I don't want to create an add-on image. The goal is not to insert the language files into the image.

Just the right way to install the templates in German.

Think it's really easier to replicate the templates (mail, discussion) from a client in German.

**Comment by Daniel-Nashed** [comment]
This isn't what you do in a container environment! You always want to install all software at build time! 
The LP needs root permission and a stopped server. 

German would be the first LP I would implement. Not only because I am German, but also because the requests mainly came from Germany.

I am not against it, but it needs still some work. 

If the use case is an existing environment, I would centralize the deployment.

**Comment by smejkal** [comment]
> This isn't what you do in a container environment! You always want to install all software at build time! 

I would agree if the goal is to install the binaries in an additional language. 

But not if want to install the templates in additional language. In my opinion, the templates belong to the user data.

On the other hand, it would also have an advantage if you can include the templates in the image. Is it possible to split the data and template directory?

**Comment by Daniel-Nashed** [comment]
the data directory from the installer is packed and included in the container image.
when you update to a new image with a different Domino version, the templates an other install data are replaced, like in any other install.

this makes the container a very similar experience, than what you see in standard servers.

you can use a data.zip which is expanded at first setup.
but there is currently no concept adding any data after initial configuration.

it's really hard to tell what the best approach is, if you are only providing part of the information I would need to understand your exact use case.

as explained I would use Domino standard deployment technology for additional templates.

you deploy them on a hub server and replicate them to the places where you need them.

a container server is nothing different than a normal server .

the only real reason for a full international template support in the container image would be a larger scale deployment with German LP. 

including it, could save time and make the deployment more standardized.

in your case it sounds like my outlined approach would work well.

personally I don't really like LPs. I recommend to customers to keep system databases in English and just replicate the localized templates from a central server.

container deployment functionality isn't the answer to every question.

"if you only have a hammer, every problem looks like a nail"

I would always look into the tool box and see what the best concept is -- which might be different depending on the Domino environment.

**Comment by Daniel-Nashed** [comment]
I spent time today to get a first version of the Domino 12.0.2 Language Pack added to the develop branch.
It was more tricky than expected. I recorded a silent install properties file and it turned out I also need a LpSilent.txt file, which needs to be specified.

The current mode is to replace all templates. I think this is what most customers want. 
Maybe we should copy the original English templates to a sub folder?
At least the English mail template would be probably useful.
